Remotion/Web/ClientScript/Scripts/DropDownMenu.ts

```diff
@@ -0,0 +1,934 @@
+
+      $(anchor).attr(itemInfo.DiagnosticMetadataForCommand);
+    }
+
+    var span = document.createElement('span');
+    span.innerHTML = itemInfo.Text!;
+    anchor.appendChild(span);
+    $(anchor)
+      .bind ('focus', DropDownMenu.FocusHandler)
+      .bind ('blur', DropDownMenu.BlurHandler);
+
+    return item;
+  }
+
+  private static CreateSeparatorItem(): HTMLLIElement
+  {
+    var item = document.createElement('li');
+
+    var textPane = document.createElement('span');
+    textPane.className = DropDownMenu._itemSeparatorClassName;
+    item.setAttribute('role', 'none');
+    textPane.innerHTML = '&nbsp;';
+
+    item.appendChild(textPane);
+
+    return item;
+  }
+
+  public static OnHeadControlClick(): void
+  {
+    DropDownMenu._itemClicked = true;
+  }
+
+  public static OnKeyDown(event: JQueryKeyEventObject, dropDownMenu: JQuery, getSelectionCount: Nullable<DropDownMenu_SelectionCountGetter>, hasDedicatedDropDownMenuElement: boolean): void
+  {
+    ArgumentUtility.CheckNotNullAndTypeIsJQuery('dropDownMenu', dropDownMenu);
+
+    if (DropDownMenu._currentMenu === null && !hasDedicatedDropDownMenuElement)
+      return;
+
+    switch (event.keyCode)
+    {
+      case 9: // tab
+        DropDownMenu.ClosePopUp(DropDownMenu._updateFocus);
+        return;
+      case 13: //enter
+      case 32: //space
+        event.preventDefault();
+        event.stopPropagation();
+        if (dropDownMenu !== DropDownMenu._currentMenu)
+        {
+          DropDownMenu.OnClick (dropDownMenu, dropDownMenu[0].id, getSelectionCount, null);
+          event.keyCode = 40; // always act as if the down-arrow was used when opening the drop down menu.
+          DropDownMenu.Options_OnKeyDown (event, dropDownMenu);
+        }
+        else
+        {
+          DropDownMenu.ClosePopUp(DropDownMenu._updateFocus);
+        }
+        return;
+      case 27: //escape
+        event.preventDefault();
+        event.stopPropagation();
+        DropDownMenu.ClosePopUp(DropDownMenu._updateFocus);
+        return;
+      case 38: // up arrow
+      case 40: // down arrow
+        event.preventDefault();
+        event.stopPropagation();
+        if (dropDownMenu !== DropDownMenu._currentMenu)
+        {
+          DropDownMenu.OnClick (dropDownMenu, dropDownMenu[0].id, getSelectionCount, null);
+          event.keyCode = 40; // always act as if the down-arrow was used when opening the drop down menu.
+        }
+        DropDownMenu.Options_OnKeyDown (event, dropDownMenu);
+        return;
+      default:
+        DropDownMenu.Options_OnKeyDown(event, dropDownMenu);
+        return;
+    }
+  }
+
+  private static Options_OnKeyDown(event: JQueryKeyEventObject, dropDownMenu: JQuery): void
+  {
+    ArgumentUtility.CheckNotNullAndTypeIsJQuery('dropDownMenu', dropDownMenu);
+
+    if (DropDownMenu._currentPopup == null)
+      return;
+
+    // TODO: bug?
+    var itemInfos = DropDownMenu._menuInfos[dropDownMenu[0].id]!.ItemInfos!;
```

Comment:
both values cannot be null when this code is reached. There's probably not much we can do to make this typesafe right now. Maybe a discriminated union down the road?

Reply:
@MichaelKetting  So I should remove the TODO and leave the `!`'s as they are now?

Reply:
discuss with @a-ctor if a follow-up ticket has any chance of yielding a usable solution

Reply:
@iam-mholle decide after checking with @a-ctor: if there's a good idea on how to improve it, we can add a todo with issue reference. otherwise, just drop the todo

Reply:
We already have a "discriminated union" here: `DropDownMenu_ItemInfo[] | null`. We should just check for the nullablility here as well.

Another solution would be to remove the nullability and use an empty array instead of `null`.

Reply:
@a-ctor I was thinking more along the lines that the implementation doesn't actually allow the null-cases to be triggered by the current code. So, maybe there's a way to redesign the entire file to make this null-safe. The only question is, if this is even sensible because it's definitely too much effort to do this right now. => The dropdownmenu must be in the collection and must have ItemInfos, otherwise, the keydown-event could not have been fired in the first place. 

cc @iam-mholle

Reply:
Result of a discussion with MK: Remove todo and leave with both `!`. Both should never be null in this context and this preserves the previous behavior of throwing an error if one of the targets is `null`.